### PR TITLE
OptimizeIfChains added

### PR DESCRIPTION
**Pull request by kamalov-ruslan**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Performance Improvement

Changelog entry (up to few sentences, required except for Non-significant/Documentation categories):
If chains are now optimized as multiIf.

### CODE REVIEW

File: dbms/src/Interpreters/OptimizeIfChains.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#ifndef CLICKHOUSE_OPTIMIZEIFCHAINS_H
```
Comment (alexey-milovidov) [inline dbms/src/Interpreters/OptimizeIfChains.h]: We don't use include guards, only #pragma once.

Comment (4ertus2) [inline dbms/src/Interpreters/OptimizeIfChains.h]: #pragma once

File: dbms/src/Interpreters/OptimizeIfChains.cpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+#include <Parsers/ASTLiteral.h>
+#include <Parsers/ASTFunction.h>
+#include <Parsers/ASTExpressionList.h>
+#include <Interpreters/OptimizeIfChains.h>
+#include <IO/WriteHelpers.h>
+
+namespace DB
+{
+
+    namespace ErrorCodes
```
Comment (4ertus2) [inline dbms/src/Interpreters/OptimizeIfChains.cpp]: We do not add tab for DB namespace content.

File: dbms/src/Interpreters/OptimizeIfChains.cpp [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                continue;
+            }
+
+            auto chain = if_chain(child);
+            reverse(chain.begin(), chain.end());
+            child = makeASTFunction("multiIf", std::move(chain));
+        }
+    }
+
+    ASTs OptimizeIfChainsVisitor::if_chain (ASTPtr & child) {
```
Comment (4ertus2) [inline dbms/src/Interpreters/OptimizeIfChains.cpp]: coding style: `ifChain()`. Camel case, no space after fucntion name, brase on its own line.

**Review by 4ertus2 — APPROVED** [review]
LGTM

File: dbms/src/Interpreters/OptimizeIfChains.cpp [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+{
+    if (!current_ast)
+    {
+        return;
+    }
+    for (ASTPtr & child : current_ast->children)
+    {
+        auto * function_node = child->as<ASTFunction>();
+
+        if (!function_node || function_node->name != "if" ||
```
Comment (4ertus2) [inline dbms/src/Interpreters/OptimizeIfChains.cpp]: Would it better to rewrite tis code with
```
if (function_node && function_node->name == "if" ...)
{
...
}
else
{
            visit(child);
￼            continue;
}
```

**Pull request merged**